sonic-xcvrd/xcvrd/xcvrd.py

```diff
@@ -1502,6 +1688,20 @@ def task_worker(self):
 
                         self.log_notice("{}: READY".format(lport))
                         self.port_dict[lport]['cmis_state'] = self.CMIS_STATE_READY
+
+                        # Setting CMIS_REINIT_REQUIRED to false to avoid CMIS re-init caused by TRANSCEIVER_INFO table
+                        # update through post_port_active_apsel_to_db
+                        asic_id = self.port_mapping.get_asic_id_for_logical_port(lport)
+                        cmis_reinit_rqd_val = get_state_db_port_table_val_by_key(
+                                        self.xcvr_table_helper, self.port_mapping, lport, "CMIS_REINIT_REQUIRED")
+                        if cmis_reinit_rqd_val is None:
+                            self.log_error("{}: CMIS_STATE_DP_ACTIVATE state: cmis_reinit_rqd_val is None".format(lport))
+                        else:
+                            if cmis_reinit_rqd_val == "true":
+                                state_port_table = self.xcvr_table_helper.get_state_port_tbl(asic_id)
+                                state_port_table.set(lport, [("CMIS_REINIT_REQUIRED", "false")])
```

Comment:
Not sure if `CMIS_REINIT_REQUIRED` is the best flag/naming, it seems to be ambiguous since re-init can also be required in cases such as error/timeout/host_tx_ready_change/etc. 
Moreover, `force_cmis_reinit()` can get triggered at any time if the listened DB contents get touched/`SET` , please refer to below code snippet (though not sure if the code is doing the correct thing by expanding the coverage of re-init to so many cases): https://github.com/sonic-net/sonic-platform-daemons/blob/70e875e5f21d408bedc4251164f1898c48c205ea/sonic-xcvrd/xcvrd/xcvrd.py#L912-L930


There might be a better flag/naming. (maybe `CMIS_INIT_DONE`/`CMIS_DONE`/`XXX_DONE`/etc is less ambiguous? considering `DONE` can only be `True` when state machine goes to certain final state)